lib/connector-extension/connector-extension-client.ts

```diff
@@ -0,0 +1,201 @@
+import { Subjects } from './subjects'
+import { AppLogger } from '../helpers/logger'
+import {
+  CallbackFns,
+  IncomingMessage,
+  WalletInteraction,
+  WalletInteractionSuccessResponse,
+  eventType,
+} from '../IO/v1/schemas'
+import { Err, Result, ResultAsync, err, ok } from 'neverthrow'
+import { SdkError, createSdkError } from '../wallet-sdk'
+import {
+  Subject,
+  Subscription,
+  filter,
+  first,
+  firstValueFrom,
+  map,
+  merge,
+  of,
+  share,
+  takeUntil,
+  tap,
+  timer,
+} from 'rxjs'
+import { unwrapObservable } from '../helpers/unwrap-observable'
+import { config } from '../config'
+
+export type ConnectorExtensionClient = ReturnType<
```

Comment:
I saw someone complained about us using `ReturnType` so often. Do you think it'd bring any more value if you did explicit typing here?